test/tools/utils.tests.js

```diff
@@ -159,20 +178,146 @@ describe('#utils calcChanges', () => {
     ];
     const assets = [ { name: 'Name1', id: 'id1' } ];
 
-    const { update } = utils.calcChanges(assets, existing, [ 'id', 'name' ]);
+    const { update } = utils.calcChanges(mockHandler, assets, existing, [ 'id', 'name' ]);
 
     expect(update).to.have.length(1);
     expect(update).to.deep.include({ name: 'Name1', id: 'id1' });
   });
 
+  it('should skip deletion of dropped keys in objectFields when ALLOW_DELETE is not enabled', () => {
+    const existing = [
+      {
+        id: 'id3',
+        metadata: {
+          should_be_preserved: 'should_be_preserved',
+          should_be_changed: 'should_be_changed',
+          should_be_removed: 'should_be_removed'
+        }
+      }
+    ];
+    const assets = [
+      {
+        id: 'id3',
+        metadata: {
+          should_be_preserved: 'should_be_preserved',
```

Comment:
Maybe remove this one, if we want to test orphans metadata.

Reply:
The test for orphans is the `should_be_removed` value in the `existing` array. Since it's not in the `assets` array, it  is simulating a property set in the dashboard but not reflected in the manifest.

Reply:
`should_be_removed` got me confused, I thought it would be deleted, but it's intended to be remove from the diff.